File: 2021/May/Day 12/luckyNumberInMatrix.py
# Source :https://leetcode.com/problems/lucky-numbers-in-a-matrix/submissions/
class Solution(object):
    def luckyNumbers (self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: List[int]
        """
        row_min =[float("inf") for _ in range(len(matrix))]
        col_max =[float("-inf") for _ in range(len(matrix[0]))]

        lists =[];

        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if(matrix[i][j]<row_min[i]):
                    row_min[i] = matrix[i][j]
                if(matrix[i][j]>col_max[j]):
                    col_max[j] = matrix[i][j]

        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if(matrix[i][j] == row_min[i]) and (matrix[i][j]==col_max[j]):
                    lists.append(matrix[i][j])
        return lists

File: 2023/January/Day 24/luckyNumberInMatrix.cpp
// Source : https://leetcode.com/problems/lucky-numbers-in-a-matrix/submissions/
// Potential Upgrade : Save these in a list so as to quickly refer instead of finding all time.
class Solution {
public:
    
    
    int findMinRow(vector<vector<int>>& matrix,int i,int j)
    {
        int min=INT_MAX;
        for(int j=0;j<size(matrix[0]);j++)
        {
            if(matrix[i][j]<min)min=matrix[i][j];
        }
        return min;
        
    }
    
    
        int findMaxCol(vector<vector<int>>& matrix,int i,int j)
    {
        int max=0;
        for(int i=0;i<size(matrix);i++)
        {
            if(matrix[i][j]>max)max=matrix[i][j];
        }
        return max;
        
    }
    
    
    
    
    
    
    
    vector<int> luckyNumbers (vector<vector<int>>& matrix) {
       
        vector<int> out;
        for(int i=0;i<size(matrix);i++)
        {
            for(int j=0;j<size(matrix[0]);j++)
            {
                
                
                if(findMinRow(matrix,i,j) == findMaxCol(matrix,i,j))
                {
                    out.insert(out.end(),matrix[i][j]);
                }
                
            }
        }
        return out;
        
        
    }
};